### PR TITLE
Merging in upstream changes

### PR DESCRIPTION
**Pull request by P3TE**
Merging in upstream changes.
There's also some other small changes in UrdfJointDescription.cs and the class 'Axis'.

### CODE REVIEW

File: com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfGeometryCollision.cs [hunk 1]
```diff
@@ -307,25 +319,43 @@ private static void ConvertMeshToColliders(GameObject gameObject, string locatio
                     List<Mesh> colliderMeshes = decomposer.GenerateConvexMeshes(meshFilter.sharedMesh);
                     foreach (Mesh collider in colliderMeshes)
                     {
+                        var c = collider;
```
Comment (Chootin) [inline com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfGeometryCollision.cs]: Is it strictly necessary to declare a new variable here?

File: com.unity.robotics.urdf-importer/Runtime/RosSharpDefinitions/UrdfJointDescription.cs [hunk 2]
```diff
@@ -95,45 +96,66 @@ public void WriteToUrdf(XmlWriter writer)
...
-            public static Axis DefaultAxis => new Axis(new double[] {1, 0, 0});
+            public Vector3 axisROS;
+            
+            const string _AtrributeName = "xyz";
 
-            public Axis(XElement node)
+            private static Vector3 GetDefaultAxisValues()
             {
-                xyz = node.Attribute("xyz") != null ? node.Attribute("xyz").ReadDoubleArray() : null;
+                return new Vector3(1, 0, 0);
```
Comment (Chootin) [inline com.unity.robotics.urdf-importer/Runtime/RosSharpDefinitions/UrdfJointDescription.cs:107]: Should we just use Vector.right here? It's perhaps a little harder to read - does it describe better what's actually going on?

File: com.unity.robotics.urdf-importer/Runtime/RosSharpDefinitions/UrdfJointDescription.cs [hunk 3]
```diff
@@ -95,45 +96,66 @@ public void WriteToUrdf(XmlWriter writer)
...
             }
 
-            public Axis(double[] xyz)
+            private void SetAxisRos(Vector3 newAxisRos)
             {
-                this.xyz = xyz;
+                if (newAxisRos.sqrMagnitude < 0.001f)
+                {
+                    RuntimeUrdf.AddImportWarning($"Axis attribute '{_AtrributeName}' magnitude should be non zero! Found: {newAxisRos}, using default axis.");
```
Comment (Chootin) [inline com.unity.robotics.urdf-importer/Runtime/RosSharpDefinitions/UrdfJointDescription.cs]: _AttributeName

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfComparator.cs [hunk 4]
```diff
@@ -522,6 +522,21 @@ private bool CompareGeometry( UrdfLinkDescription.Geometry source,  UrdfLinkDesc
...
+            {
+                bool capsuleEqual = (source.capsule.radius == exported.capsule.radius && source.capsule.length == exported.capsule.length);
+                linkLog.AppendLine(String.Format("{0}Geometry:", Indent(indent)));
+                linkLog.AppendLine(String.Format("{0}Type:{1,5}", Indent(indent), "Capsule"));
+                linkLog.AppendLine(String.Format("{0}Dimensions Equal: {1,6}", Indent(indent), capsuleEqual));
+                linkLog.AppendLine(String.Format("{0}Source: Radius: {1,5:F3} Length: {2,5:F3} Exported: Radius: {3,5:F3} Length: {4,5:F3}", Indent(indent), source.capsule.radius, source.capsule.length, exported.capsule.radius, exported.capsule.length));
+
+                if (capsuleEqual)
+                    return true;
+
```
Comment (Chootin) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfComparator.cs]: Spacing here is weird. Also consider just using explicit braces.

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs [hunk 5]
```diff
@@ -52,60 +59,75 @@ public Vector3 AdjustedCenterOfMass
...
+        public void ResetInertial()
+        {
+            m_Overrides = m_OriginalValues;
+            ImportInertiaData(m_Overrides);
+        }
+
 #region Runtime
 
-        private void Start()
+        void Start()
```
Comment (Chootin) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfInertial.cs]: Why not private?

File: com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointFloating.cs [hunk 6]
```diff
@@ -37,6 +37,7 @@ public static UrdfJoint Create(GameObject linkObject)
         public override float GetPosition()
         {
             #if  UNITY_2020_1_OR_NEWER && !URDF_FORCE_RIGIDBODY
+                // not true, weird - talk to Vidur
                 Debug.Log("'ArticulationBody' does not contain a definition for 'connectedAnchor' and no accessible extension method 'connectedAnchor'");
                 Vector3 distanceFromAnchor = ((ArticulationBody)unityJoint).transform.localPosition ;/*-
```
Comment (Chootin) [inline com.unity.robotics.urdf-importer/Runtime/UrdfComponents/UrdfJoints/UrdfJointFloating.cs:42]: Look into this.

**Review by Chootin — CHANGES_REQUESTED** [review]
See comments.

File: com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfGeometryCollision.cs (hunk 1 above)
Comment (P3TE) [reply com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfGeometryCollision.cs]: Foreach iteration variable 'collider' is immutable.
So yes, apparently.
I'm going to go ahead and swap 'c' and 'collider' around however.

File: com.unity.robotics.urdf-importer/Runtime/RosSharpDefinitions/UrdfJointDescription.cs (hunk 2 above)
Comment (P3TE) [reply com.unity.robotics.urdf-importer/Runtime/RosSharpDefinitions/UrdfJointDescription.cs:107]: Added a comment. I think it should be kept as new Vector3(1, 0, 0);

**Pull request merged**